packages/labs/react/src/create-component.ts

```diff
@@ -122,15 +122,22 @@ const setProperty = <E extends Element>(
   events?: EventNames
 ) => {
   const event = events?.[name];
-  if (event !== undefined) {
+  if (event !== undefined && value !== old) {
     // Dirty check event value.
-    if (value !== old) {
-      addOrUpdateEventListener(node, event, value as (e?: Event) => void);
-    }
-  } else {
-    // But don't dirty check properties; elements are assumed to do this.
-    node[name as keyof E] = value as E[keyof E];
+    addOrUpdateEventListener(node, event, value as (e?: Event) => void);
+    return;
   }
+
+  if (
```

Comment:
```suggestion
// Note, the attribute removal here for `undefined` and `null` values is done
// to match React's behavior on non-custom elements. It needs special 
// handling because it does not match platform behavior.  For example, 
// setting the `id` property to `undefined` sets the attribute to the string
// "undefined." React "fixes" that odd behavior and the code here matches
// React's convention.
  if (
```

Reply:
copy pasted into commit